src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -1110,11 +1110,40 @@ bool entitiesQuery
     }
 
     /* Process query result */
-    while (cursor->more()) {
+    while (cursor->more())
+    {
+        BSONObj                 r    = cursor->next();
+        std::string             err  = r.getStringField("$err");
+        ContextElementResponse* cer  = new ContextElementResponse();
+
+        if (err != "")
+        {
+          //
+          // We can't return the error 'as is', as it may contain forbidden characters.
```

Comment:
Good point. However, two things to take into account:

1. We have a rendering logic that scape forbidden characteres, even if they are produced by a "third party" as the DB.
2. The exact message that MongoDB reports as error could vary from version to version (e.g. from MongoDB 2.4 to MongoDB 2.6), which could cause this mechanism change and some confusion.

Having said that, I'm not sure what is the best mechanism (either as it is now in the code or just passing the message returned by MongoDB)...